homeassistant/components/fritz/binary_sensor.py

```diff
@@ -17,79 +24,91 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def _retrieve_connection_state(status: FritzStatus) -> bool:
+    """Return download line attenuation."""
+    return bool(status.is_connected)
+
+
+def _retrieve_link_state(status: FritzStatus) -> bool:
+    """Return download line attenuation."""
+    return bool(status.is_linked)
+
+
+@dataclass
+class SensorData(SensorEntityDescription):
+    """Fritz sensor data class."""
+
+    state_provider: Callable | None = None
```

Comment:
If you remove the `state_provider` functions per my comment, this custom description would no longer be needed

Reply:
Class removed